src/app/containers/Figure/index.jsx

```diff
@@ -1,37 +1,59 @@
 import React from 'react';
 import { string, number, objectOf, any } from 'prop-types';
 import Figure from '../../components/Figure';
-import Image from '../../components/Figure/Image';
+import Image, { AmpImg } from '../../components/Figure/Image';
 import ImagePlaceholder from '../../components/Figure/ImagePlaceholder';
 import Copyright from '../Copyright';
 import Caption from '../Caption';
+import { PlatformContextConsumer } from '../../contexts/PlatformContext';
 
 const renderCopyright = copyright =>
   copyright ? <Copyright>{copyright}</Copyright> : null;
 
 const renderCaption = block => (block ? <Caption block={block} /> : null);
 
-const FigureContainer = ({ src, alt, ratio, copyright, captionBlock }) => (
-  <Figure>
-    <ImagePlaceholder ratio={ratio}>
-      <Image alt={alt} src={src} />
-      {renderCopyright(copyright)}
-    </ImagePlaceholder>
-    {renderCaption(captionBlock)}
-  </Figure>
+const renderImage = (platform, alt, src, height, width) => {
+  if (platform === 'amp') {
+    return <AmpImg alt={alt} src={src} height={height} width={width} />;
+  }
+  return <Image alt={alt} src={src} height={height} width={width} />;
+};
+
+const FigureContainer = ({
+  alt,
+  captionBlock,
+  copyright,
+  height,
+  ratio,
+  src,
+  width,
+}) => (
+  <PlatformContextConsumer>
+    {platform => (
+      <Figure>
+        <ImagePlaceholder ratio={ratio}>
+          {renderImage(platform, alt, src, height, width)}
+          {renderCopyright(copyright)}
+        </ImagePlaceholder>
+        {renderCaption(captionBlock)}
+      </Figure>
+    )}
+  </PlatformContextConsumer>
 );
 
 FigureContainer.propTypes = {
   alt: string.isRequired,
-  src: string.isRequired,
-  ratio: number.isRequired,
-  copyright: string,
   captionBlock: objectOf(any),
+  copyright: string,
+  height: number.isRequired,
```

Comment:
let's move height down for ordering

Reply:
The aim is to alphabetise the props, for readability.

Reply:
Hmm - I see why you did that. But I think it's like CSS properties, we ought to group by type rather than alphabetically:

![45% CSS devs write by type vs 14% alphabetical](https://css-tricks.com/wp-content/uploads/2012/08/orgchart.png)